src/Microsoft.OData.Core/Uri/ApplyClauseToStringBuilder.cs

```diff
@@ -210,17 +210,7 @@ private void Translate(GroupByTransformationNode transformation)
                     query.Append(ExpressionConstants.SymbolOpenParen);
                 }
 
-                if (node.Expression != null)
-                {
-                    AppendExpression(node.Expression);
-                }
-
-                bool appendCommaChild = false;
-                foreach (GroupByPropertyNode childNode in node.ChildTransformations)
-                {
-                    appendCommaChild = AppendComma(appendCommaChild);
-                    AppendExpression(childNode.Expression);
-                }
+                Translate(node);

```

Comment:
Could you add test to ApplyBuilderTest.cs to validate that change?

Reply:
Added some more tests. The covered scenarios are the same like the ones for the `ApplyBinder`.

Btw, I noticed something that might be considered as a bug:

If I add the following test case for the `ApplyClauseToStringBuilder`:
_groupby((MyDog/Color,MyAddress/City,MyDog/Breed))_ , the expected result is _groupby((MyDog/Color,MyDog/Breed,MyAddress/City))_ and I think that is due to [the RegisterProperty method in ApplyBinder](https://github.com/OData/odata.net/blob/7b446f94132ca9f28d7e8a79954692519c08a386/src/Microsoft.OData.Core/UriParser/Aggregation/ApplyBinder.cs#L312) . It's trying to find a container (building a tree basically) for each part of the path (non-leaf node) and the order cannot be traced later.